lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3281,6 +3281,17 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       // Record the body.
       TC.definedFunctions.push_back(FD);
     }
+    
+    // If any of the function's parameters contain an uninhabited type
+    // then emit a diagnostic on the func decl, as the function can never
+    // be called.
+    auto params = FD->getParameters()->getArray();
+    for (auto &&index: indices(params)) {
+      auto param = params[index];
+      if (param->hasInterfaceType() && param->getType()->isStructurallyUninhabited()) {
```

Comment:
getInterfaceType() not getType() since its cheaper

Reply:
I see! Fixed now.